### PR TITLE
feat: add transfer workflows function with l2 signer

### PR DESCRIPTION
**Pull request by yundi-fu**
# Summary
- added `batchTransfersWorkflowWithSigner`
- added ` transferWithSigner`

# Why the changes

# Things worth calling out

### CODE REVIEW

File: src/workflows/transfers.ts [hunk 1]
```diff
@@ -5,10 +5,75 @@ import {
   GetSignableTransferRequestV1,
   CreateTransferResponseV1,
   GetSignableTransferRequest,
-  CreateTransferResponse,
+  CreateTransferResponse, SignableTransferResponseDetails,
```
Comment (ABashkirtseva) [inline src/workflows/transfers.ts:8]: Can you please remove `SignableTransferResponseDetails`

**Review by ABashkirtseva — APPROVED** [review]

File: src/workflows/workflows.ts [hunk 2]
```diff
@@ -121,6 +127,12 @@ export class Workflows {
     return transfersWorkflow(signer, starkWallet, request, this.transfersApi);
   }
 
+  public transferWithSigner(walletConnection: WalletConnection, request: UnsignedTransferRequest) {
+    return transfersWorkflowWithSigner(
+      { ...walletConnection, request, transfersApi: this.transfersApi  });
```
Comment (carmen0208) [inline src/workflows/workflows.ts:132]:
```suggestion
      { ...walletConnection, request, transfersApi: this.transfersApi });
```

File: src/workflows/transfers.ts [hunk 3]
```diff
@@ -123,3 +188,66 @@ export async function batchTransfersWorkflow(
...
   };
 }
+
+
+export async function batchTransfersWorkflowWithSigner({
+  l1Signer,
+  l2Signer,
+  request,
+  transfersApi,
+}:BatchTransferRequestParams): Promise<CreateTransferResponse> {
```
Comment (carmen0208) [inline src/workflows/transfers.ts:198]:
```suggestion
}: BatchTransferRequestParams): Promise<CreateTransferResponse> {
```

File: src/workflows/transfers.ts [hunk 4]
```diff
@@ -123,3 +188,66 @@ export async function batchTransfersWorkflow(
...
+  });
+
+  const signableMessage = signableResult.data.signable_message;
+
+  if (signableMessage === undefined) {
+    throw new Error('Invalid response from Signable registration offchain');
+  }
+
+  // Obtain Ethereum Address from signer
+  const ethAddress = (await l1Signer.getAddress());
```
Comment (carmen0208) [inline src/workflows/transfers.ts:214]:
```suggestion
  const ethAddress = await l1Signer.getAddress();
```

**Review by carmen0208 — APPROVED** [review]

**Review by zatlite — APPROVED** [review]

**Pull request merged**